app/models/merchant.rb

```diff
@@ -4,4 +4,14 @@ class Merchant < ApplicationRecord
   has_many :transactions, through: :invoices
 
   validates :name, presence: true
+
+  def total_revenue
+    invoices.select(&:successful?).reduce(0) do |total, invoice|
+      total += invoice.total
+    end
+  end
+
+  def self.with_invoices
```

Comment:
I like the ideas here, but do recommend more AR and less vanilla Ruby. It seems to me that what you're trying to return is a collection of Invoice objects that have a successful transaction. You can start with that `includes` call, and then scope the result down using `.where`.

Remember, `.where` can take raw SQL as an argument, so you should be able to specify a particular result on `transactions` (there's also a way to do this with a nested hash, but the SQL might be more straightforward?). We can definitely talk through this more tomorrow.